src/main/java/com/arpnetworking/metrics/impl/BaseFileSink.java

```diff
@@ -441,7 +442,7 @@ protected void validate(final List<String> failures) {
         protected Boolean _prudent = DEFAULT_PRUDENT;
         protected Boolean _dropWhenQueueFull = DEFAULT_DROP_WHEN_QUEUE_FULL;
         protected Integer _maxQueueSize = DEFAULT_MAX_QUEUE_SIZE;
-        protected String _maxFileSize = DEFAULT_MAX_FILE_SIZE;
+        protected String _maxFileSize = null;
         protected String _totalSizeCap = DEFAULT_TOTAL_SIZE_CAP;
```

Comment:
As discussed I think the total cap size should be function of the max history and the file size if set. If there is not max file size then there is no total cap. After thinking about this further, I think it would be easier to migrate to the Logback Steno random size and time rotation.